gui/src/main/java/org/verapdf/processor/ProcessorImpl.java

```diff
@@ -0,0 +1,276 @@
+
+	@Override
+	public ProcessingResult validate(InputStream pdfFileStream, ItemDetails fileDetails,
+						 Config config, OutputStream reportOutputStream) {
+		ValidationResult validationResult = null;
+		MetadataFixerResult fixerResult = null;
+		FeaturesCollection featuresCollection = null;
+		ProcessingResult processingResult = new ProcessingResult(config);
+
+		PDFAValidator validator;
+		ValidationProfile validationProfile;
+
+		try {
+			validationProfile = profileFromConfig(config);
+		} catch (JAXBException e) {
+			LOGGER.error("Error in parsing profile XML", e);
+			processingResult.getErrorMessages().add(
+					"Error in parsing profile from XML: " + e.getMessage());
+			processingResult.setErrorInValidation();
+			processingResult.setErrorInMetadataFixer();
+			validationProfile = Profiles.defaultProfile();
+		} catch (IOException e) {
+			LOGGER.error("Error in reading profile from disc", e);
+			processingResult.getErrorMessages().add(
+					"Error in reading profile from disc: " + e.getMessage());
+			processingResult.setErrorInValidation();
+			processingResult.setErrorInMetadataFixer();
+			validationProfile = Profiles.defaultProfile();
+		}
+		validator = (validationProfile == Profiles.defaultProfile())
+				? null : Validators.createValidator(validationProfile,
+				logPassed(config), config.getMaxNumberOfFailedChecks());
+		long startTimeOfValidation = System.currentTimeMillis();
+		try (ModelParser toValidate = new ModelParser(pdfFileStream,
+				validationProfile.getPDFAFlavour())) {
+
+			if(config.getProcessingType().isValidating()) {
+				if (validator != null) {
+					try {
+						validationResult = validator.validate(toValidate);
+						if (!validationResult.isCompliant()) {
+							processingResult.setValidationSummary(
+									ProcessingResult.ValidationSummary.FILE_NOT_COMPLIANT);
+						}
+					} catch (IOException | ValidationException e) {
+						LOGGER.error("Error in validation", e);
+						processingResult.setErrorInValidation();
+						processingResult.setErrorInMetadataFixer();
+						processingResult.getErrorMessages().add(
+								"Error in validation: " + e.getMessage());
+					}
+
+					if (config.isFixMetadata() && validationResult != null) {
+						try {
+							fixerResult = fixMetadata(validationResult, toValidate,
+									fileDetails.getName(), config);
+						} catch (IOException e) {
+							LOGGER.error("Error in fixing metadata", e);
+							processingResult.setErrorInMetadataFixer();
+							processingResult.getErrorMessages().add(
+									"Error in fixing metadata: " + e.getMessage());
+						}
+					}
+				}
+			}
+			if (config.getProcessingType().isFeatures()) {
+				try {
+					featuresCollection = PBFeatureParser
+							.getFeaturesCollection(toValidate.getPDDocument());
+				} catch (Exception e) {
+					LOGGER.error("Error in extracting features", e);
+					processingResult.setErrorInFeatures();
+					processingResult.getErrorMessages().add("Error in feature extraction: "
```

Comment:
add a method "addErrorMessage" into processing result instead of getting a list and adding to it